src/OrbitGl/PrimitiveAssemblerTest.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "MockBatcher.h"
+#include "PickingManagerTest.h"
+#include "PrimitiveAssembler.h"
+
+namespace orbit_gl {
+
+// TODO(http://b/228063067): Test all methods in Primitive Assembler
+class PrimitiveAssemblerTester : public PrimitiveAssembler {
+ public:
+  PrimitiveAssemblerTester(PickingManager* picking_manager = nullptr)
+      : PrimitiveAssembler(&mock_batcher_, picking_manager) {}
+  void ExpectAdded(uint32_t line_count, uint32_t triangle_count, uint32_t box_count) {
+    EXPECT_EQ(mock_batcher_.GetNumLines(), line_count);
+    EXPECT_EQ(mock_batcher_.GetNumTriangles(), triangle_count);
+    EXPECT_EQ(mock_batcher_.GetNumBoxes(), box_count);
+  }
+
+ private:
+  MockBatcher mock_batcher_;
+};
+
+TEST(PrimitiveAssembler, NullPickingManager) {
+  PrimitiveAssemblerTester primitive_assembler_tester;
+  std::shared_ptr<PickableMock> pickable = std::make_shared<PickableMock>();
+  EXPECT_DEATH(primitive_assembler_tester.AddLine(Vec2(0, 0), Vec2(1, 0), 0,
+                                                  Color(255, 255, 255, 255), pickable),
+               "nullptr");
+}
+
+static Vec3 Vec2ToVec3(Vec2 vertex, float z = 0) { return {vertex[0], vertex[1], z}; }
```

Comment:
pers pref: Have all helper methods before the actual TESTs

Reply:
Will move it to CoreMath.h class